src/app/region.js

```diff
@@ -178,9 +182,25 @@ export function showRegion (bounds) {
  *            OSMLR geometry tiles, merged into a single GeoJSON.
  */
 function fetchOSMLRGeometryTiles (suffixes) {
-  const urls = suffixes.map(suffix => `${OSMLR_TILE_PATH}${suffix}.json`)
-  const fetchTiles = urls.map(url => fetch(url).then(res => res.json()))
-  // Results is an array of all GeoJSON tiles. Next, merge into one file
-  // and return the result as a single GeoJSON.
-  return Promise.all(fetchTiles).then(merge)
+  const tiles = suffixes.map(suffix => fetchOSMLRtile(suffix))
+  return Promise.all(tiles).then(merge)
+}
+
+function fetchOSMLRtile (suffix) {
+  // If tile already exists in cache, return a copy of it
+  if (OSMLRCache[suffix]) {
+    const clone = JSON.parse(JSON.stringify(OSMLRCache[suffix]))
+    return clone
+  }
+  // Otherwise make a request for it, cache it, then return a copy of it
+  const url = `${OSMLR_TILE_PATH}${suffix}.json`
+  return window.fetch(url)
+    .then(results => results.json())
+    .then(res => cacheOSMLRTiles(res, suffix))
+}
+
+function cacheOSMLRTiles (tile, index) {
+  Object.assign(OSMLRCache, {[index]: tile})
+  const clone = JSON.parse(JSON.stringify(tile))
```

Comment:
Do we need to clone the tile after caching it?

Reply:
Yea for some reason, if I don't return the clone version of it then it directly changes the properties of the tile so that next time I use it, the parts of the tile that I filtered out are still filtered out and the routes don't show. 
From what I can understand after googling things is that even though I use a copy of the `features` it still contains reference to its properties so when I filter out the latlngs of a feature the original `feature` gets filtered out too. Not sure how to fix it.

Reply:
oh I see. yep, objects do that...

is it still a problem if we clone immediately after the merge in [line 186](https://github.com/opentraffic/analyst-ui/blob/d0a1c4beb5a7b5fbc587840857040e1ef2e79f9e/src/app/region.js#L186)?

Reply:
https://github.com/opentraffic/analyst-ui/pull/96/commits/1bc6c3cc08e835cbb61ff8a07fdbee779f7e7b7c

It works if I do this^. If we do it like this, would we still need the `results.features.slice()` since it's already a copy?

Reply:
Cool - I think you're right, in that case I'm pretty sure we don't need the `slice()`.